src/test/ui/issues/issue-24036.stderr

```diff
@@ -2,7 +2,10 @@ error[E0308]: mismatched types
   --> $DIR/issue-24036.rs:3:9
    |
 LL |     let mut x = |c| c + 1;
-   |                 --------- the expected closure
+   |                 ---------
+   |                 |
+   |                 the expected closure
+   |                 expected due to this value
```

Comment:
This seems unideal. Can we deduplicate in this case?

Reply:
The labels are added by two different code-paths, I'll see what I can do :)

Reply:
Fixed